src/components/CartFlyout.vue

```diff
@@ -0,0 +1,61 @@
+<script lang="ts">
+import { PropType, defineComponent } from 'vue';
+import { CartItem } from '../types';
+
+export default defineComponent({
+    props: {
+        isOpen: {
+            type: Boolean,
+            default: false,
+        },
+        cartItems: {
+            type: Array as PropType<CartItem[]>,
+            default: () => [],
+        },
+    },
+    methods: {
+        removeFromCart(itemId: number) {
+            this.$emit('remove-item', itemId);
+        },
+    },
+
+});
+</script>
+<template>
+    <div class="bg-white w-[95%] rounded-lg shadow-md flex flex-col m-auto mt-2" :class="{ 'hidden': !isOpen }">
+        <h2 class="font-bold text-lg mb-2 p-4">Cart</h2>
+        <div class="bg-gray-300 h-[1px] mb-6 w-full"></div>
+        <div v-if="cartItems.length === 0"
+            class="display flex justify-center items-center text-base font-bold text-neutral-dark-grayish-blue pb-20 pt-14">
+            Your cart is empty.
+        </div>
+        <template v-if="cartItems.length > 0">
+            <ul class="px-4 pb-4">
+                <li v-for="(item, index) in cartItems" :key="index" class="flex items-center justify-between mb-4">
```

Comment:
You could use destructuring to simplify code. It could be `v-for="({ id, name, price, quantity }) in cartItems"` and then instead of `item.name` just `name`.